input/otlp/traces_test.go

```diff
@@ -609,20 +609,56 @@ func TestRPCSpan(t *testing.T) {
 }
 
 func TestMessagingTransaction(t *testing.T) {
-	event := transformTransactionWithAttributes(t, map[string]interface{}{
-		"messaging.destination": "myQueue",
-	}, func(s ptrace.Span) {
-		s.SetKind(ptrace.SpanKindConsumer)
-		// Set parentID to imply this isn't the root, but
-		// kind==Consumer should still force the span to be translated
-		// as a transaction.
-		s.SetParentSpanID(pcommon.SpanID{3})
-	})
-	assert.Equal(t, "messaging", event.Transaction.Type)
-	assert.Empty(t, event.Labels)
-	assert.Equal(t, &modelpb.Message{
-		QueueName: "myQueue",
-	}, event.Transaction.Message)
+	for _, tc := range []struct {
+		attrs map[string]interface{}
+
+		expectedLabels     map[string]*modelpb.LabelValue
+		expectedTxnMessage *modelpb.Message
+	}{
+		{
+			attrs: map[string]interface{}{
+				"messaging.destination": "myQueue",
+			},
+			expectedLabels: nil,
+			expectedTxnMessage: &modelpb.Message{
+				QueueName: "myQueue",
+			},
+		},
+		{
+			attrs: map[string]interface{}{
+				"messaging.system": "kafka",
+			},
+			expectedLabels: map[string]*modelpb.LabelValue{
+				"messaging_system": {Value: "kafka"},
+			},
+			expectedTxnMessage: nil,
```

Comment:
[For reviewer] As demonstrated by the test case, this is a new case that doesn't exist before. When `messaging.system` or `messaging.operation` is present AND `messaging.destination` is absent, do we want event.Transaction.Message to be empty or nil? Does it make a difference?

To rephrase the question, is it ok to have `event.Transaction.Message == nil` when `event.Transaction.Type == "messaging"`?

Reply:
This PR fixes the bug that the transaction type is not set although messaging attributes are sent. 
However, the apm message model is pretty outdated compared to the otel messaging spec. So my suggestion would be to open a follow up github issue for updating the supported semconv version and then also update the messaging model definition. I'd see it outside the scope of this PR though.